server/backend/opensearch_security_plugin.ts

```diff
@@ -26,7 +28,34 @@ export default function (Client: any, config: any, components: any) {
    */
   Client.prototype.opensearch_security.prototype.authinfo = ca({
     url: {
-      fmt: '/_plugins/_security/authinfo',
+      fmt: `/_plugins/_security/authinfo`,
+      opt: {
+        [AUTH_TYPE_PARAM]: {
+          type: 'string',
+          required: false,
+        },
+      },
+      template: (requestObj) => {
+        const obj = requestObj || (requestObj = {});
+        let __p = '/_plugins/_security/authinfo';
+        const __q = []; // Array to hold query string components
+
+        // Iterate over the object properties to construct the query string
+        for (const key in obj) {
+          if (obj.hasOwnProperty(key)) {
+            // Ensure the value is a string to avoid URL encoding issues
+            const value = String(obj[key]);
+            __q.push(encodeURIComponent(key) + '=' + encodeURIComponent(value));
+          }
+        }
```

Comment:
Is this part necessary? There isn't validation on any other routes? Additionally there is some code here that we may want to test or use a helper library like lodash to do this?

Reply:
yes, without this the url is not modified to pass `?auth_type=` query param

Reply:
Ok if its necessary can you extract this out to a helper function and write a test for it?

Reply:
sure, I have addressed this in the latest commit.